cassandane/tiny-tests/JMAPContacts/card_set_vcardprops

```diff
@@ -0,0 +1,243 @@
+                          $card);
+    $self->assert_matches(qr|X-TIME-STANDALONE;VALUE=DATE-AND-OR-TIME:T12:30|,
+                          $card);
+    $self->assert_matches(qr|X-TIMESTAMP;VALUE=TIMESTAMP:2013-02-14T12:30:00Z|,
+                          $card);
+    $self->assert_matches(qr|X-BOOLEAN;VALUE=BOOLEAN:TRUE|, $card);
+    $self->assert_matches(qr|X-INTEGER-MIN;VALUE=INTEGER:-9223372036854775808|,
+                          $card);
+    $self->assert_matches(qr|X-INTEGER-MAX;VALUE=INTEGER:9223372036854775807|,
+                          $card);
+    $self->assert_matches(qr|X-INTEGER-MULTI;VALUE=INTEGER:1,2,3,4,5|, $card);
+    # XXX Last digit of the min/max float values gets rounded down by Jansson
+    $self->assert_matches(qr|X-FLOAT-MIN;VALUE=FLOAT:-9007199254740990\.|,
+                          $card);
+    $self->assert_matches(qr|X-FLOAT-MAX;VALUE=FLOAT:9007199254740990\.|,
+                          $card);
+    $self->assert_matches(qr|X-FLOAT-PI;VALUE=FLOAT:3\.14159265358979|, $card);
+    $self->assert_matches(qr|X-FLOAT-STRUCT;VALUE=FLOAT:1\.2\\;3\.4|,
+                          $card);
+    $self->assert_matches(qr|X-UTC-OFFSET;VALUE=UTC-OFFSET:-05:00|, $card);
+    $self->assert_matches(qr|X-LANG-TAG;VALUE=LANGUAGE-TAG:de|, $card);
+    $self->assert_matches(qr|group1\.X-UNKNOWN:foo|, $card);
+    $self->assert_matches(qr|PRODID:-//Example Corp.//CardDAV Client//EN|,
+                          $card);
+
+    xlog $self, "Update the contact";
+    $res = $jmap->CallMethods([
+        ['ContactCard/set', {
+            update => {
+                $id => {
+                    prodId => 'foo'
+                }
+            }
+        }, 'R1']
+    ]);
+    $self->assert_not_null($res->[0][1]{updated}{$id});
+
+    xlog $self, "Verify that vCardProps are round-tripped correctly";
+    $res = $carddav->Request('GET', $href, '',
+                             'Accept' => 'text/vcard; version=4.0');
+    $card = $res->{content};
+    $card =~ s/\r?\n[ \t]+//gs;  # unfold long properties
+    $self->assert_matches(qr|X-TEXT;VALUE=TEXT:baz|, $card);
+    $self->assert_matches(qr|X-TEXT-MULTI;VALUE=TEXT:foo,,bar|, $card);
+    $self->assert_matches(qr|X-URI;VALUE=URI:ldap://ldap.example.com/cn=foo|,
+                          $card);
+    $self->assert_matches(qr|X-DATE;VALUE=DATE:--04-12|, $card);
+    $self->assert_matches(qr|X-TIME-UTC;VALUE=TIME:12:30:00Z|, $card);
+    $self->assert_matches(qr|X-DATE-TIME;VALUE=DATE-TIME:2013-08-15T09:45:00\+01:00|,
+                          $card);
+    $self->assert_matches(qr|X-TIME-STANDALONE;VALUE=DATE-AND-OR-TIME:T12:30|,
+                          $card);
+    $self->assert_matches(qr|X-TIMESTAMP;VALUE=TIMESTAMP:2013-02-14T12:30:00Z|,
+                          $card);
+    $self->assert_matches(qr|X-BOOLEAN;VALUE=BOOLEAN:TRUE|, $card);
+    $self->assert_matches(qr|X-INTEGER-MIN;VALUE=INTEGER:-9223372036854775808|,
```

Comment:
Thanks for adding these! One final complaint: These kind of checks are a little too lax, it doesn't tell us if there's somehow junk after the value in the card output, it'd be good to ensure we get whatever it is we expect to be next (either \r\n, or a ;, or whatever the separator is). But I see we do this everywhere so I don't think it's necessary to fix here, but something we should fix and keep in mind when writing future tests

Reply:
Done.  I will merge when CI completes